GLPT_Cocoa.inc

```diff
@@ -1,89 +1,390 @@
+
+procedure NSShow (obj: id);
+begin
+  NSLog(obj.description, NSSTR('@'));
+end;
+
+procedure Fatal (msg: string);
+begin
+  writeln('FATAL: '+msg);
+  halt(-1);
+end;
+
+type
+  TCocoaWindow = objcclass (NSWindow)
+    ref: pGLPTwindow;
+  end;
+
+type
+  TCocoaApp = objcclass (NSApplication)
+    procedure poll; message 'poll';
+  end;
+
+function Cocoa_GetKeyboardShiftState: TShiftState; forward;
+function Cocoa_GetTime: double; forward;
+
+procedure TCocoaApp.poll;
+var
+  event: NSEvent;
+  keychar: string;
+  params: GLPT_MessageParams;
+  win: pGLPTwindow = nil;
+  pool: NSAutoreleasePool;
+begin
+  pool := NSAutoreleasePool.alloc.init;
+  event := nextEventMatchingMask_untilDate_inMode_dequeue(NSAnyEventMask, nil, NSDefaultRunLoopMode, true);
+  if event <> nil then
+    begin
+      if event.window <> nil then
+        win := TCocoaWindow(event.window).ref;
+
+      case event.type_ of
+        NSMouseMoved:
+          begin
+            params.mouse.x := trunc(event.locationInWindow.x);
```

Comment:
Is the mouse location in Mac a float? If so, shouldn't we then make the x and y variables in GLPT_MsgParmMouse also float?

Reply:
Sorry I just saw these messages (still new to GitHub).

Yeah it's technically possible to  move the mouse in sub-pixel coordinates on Mac (I would assume Windows also??) so I guess a float makes sense. However, in most cases you end up truncating the float anyways so maybe just making a another field in the record is good enough (cleans up the frequent trunc() calls).

The window rects are floats but they will never actually be < 1 so they can be truncated.